AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift

```diff
@@ -11,43 +11,219 @@
+                Log.debug(label: self.LOG_PREFIX, "Unable to dismiss \(error)")
+            }
+        }
+    }
+
+    // MARK: WKWebview delegate
+    public func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        if self.listener != nil {
+            guard let shouldOpenUrl = self.listener?.overrideUrlLoad(message: self, url: navigationAction.request.url?.absoluteString) else {
+                decisionHandler(.allow)
+                return
+            }
+            decisionHandler(shouldOpenUrl ? .allow : .cancel)
+
+        } else {
+            // if the API user doesn't provide any listner ( self.listener == nil ),
+            // set WKNavigationActionPolicyAllow as a default behaviour.
+            decisionHandler(.allow)
+        }
+    }
+
+    private func getConfiguredWebview(newFrame: CGRect) -> WKWebView {
+        let webViewConfiguration = WKWebViewConfiguration()
+
+        //Fix for media playback.
+        webViewConfiguration.allowsInlineMediaPlayback = true // Plays Media inline
+        webViewConfiguration.mediaTypesRequiringUserActionForPlayback = []
+        let wkWebView = WKWebView(frame: newFrame, configuration: webViewConfiguration)
+        wkWebView.navigationDelegate = self
+        wkWebView.scrollView.bounces = false
+        wkWebView.backgroundColor = UIColor.clear
+        wkWebView.isOpaque = false
+        wkWebView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        self.webView = wkWebView
+
+        return wkWebView
+    }
+
+    // MARK: web layout helpers
+    private func calcFullscreenFrame() -> CGRect? {
+        var newFrame = CGRect(x: 0, y: 0, width: 0, height: 0)
+        // x is always 0
+        newFrame.origin.x = 0
+        // for fullscreen, width and height are both full screen
+        let keyWindow = UIApplication.shared.getKeyWindow()
+        guard let screenBounds: CGSize = keyWindow?.frame.size else { return nil }
+        newFrame.size = screenBounds
+
+        // y is dependant on visibility and height
+        newFrame.origin.y = 0
+        return newFrame
     }
 
-    public func openUrl(url: String) {
-        fullScreenMessageHandler?.openUrl(url: url)
+    private func dismissWithAnimation(animate: Bool) {
+        DispatchQueue.main.async {
+            UIView.animate(withDuration: animate ? 0.3: 0, animations: {
+                guard var newFrame: CGRect = self.calcFullscreenFrame() else {
+                    return
+                }
+                newFrame.origin.y = newFrame.size.height
+                self.webView?.frame = newFrame
+            }) { _ in
+                self.webView?.removeFromSuperview()
+                self.webView = nil
+            }
+        }
     }
 
-    public func remove() {
-        fullScreenMessageHandler?.dismiss()
+    /// Open a url from this message
+    private func openUrl(url: String) {
```

Comment:
this method is not being used?
and if we need open url then use the `URLOpening` service

Reply:
Actually we don't need this function at all as we already have a openUrl service. In v5 we had that in fullscreen message.